odoo/service/db.py

```diff
@@ -116,12 +116,26 @@ def _create_empty_database(name):
                 sql.Identifier(name), collate, sql.Identifier(chosen_template)
             ))
 
+    # TODO: add --extension=trigram,unaccent
+    try:
+        db = odoo.sql_db.db_connect(name)
+        with db.cursor() as cr:
+            cr.execute("CREATE EXTENSION IF NOT EXISTS pg_trgm")
+    except psycopg2.Error:
+        pass
+
     if odoo.tools.config['unaccent']:
         try:
             db = odoo.sql_db.db_connect(name)
-            with closing(db.cursor()) as cr:
+            with db.cursor() as cr:
                 cr.execute("CREATE EXTENSION IF NOT EXISTS unaccent")
-                cr.commit()
+                # From PostgreSQL's point of view, making 'unaccent' immutable is incorrect
+                # because it depends on external data - see
+                # https://www.postgresql.org/message-id/flat/201012021544.oB2FiTn1041521@wwwmaster.postgresql.org#201012021544.oB2FiTn1041521@wwwmaster.postgresql.org
+                # But in the case of Odoo, we consider that those data don't
+                # change in the lifetime of a database. If they do change, all
+                # indexes created with this function become corrupted!
+                cr.execute("ALTER FUNCTION unaccent(text) IMMUTABLE")
```

Comment:
Why not do both in the same transaction / cursor? (possibly with a savepoint in case we manage to install one and not the other though that seems unlikely: both are built-in trusted extensions so the only way they would fail to install is if the current user does not have CREATE privileges, which also applies to updating a function's volatility) 

Also maybe log relevant warnings if the extensions can't be installed?

Reply:
> Why not do both in the same transaction / cursor?

You mean the same transaction than for `pg_trgm` ? Yep, it makes sense.

> Also maybe log relevant warnings if the extensions can't be installed?

Yes, good idea.